Clang build fails

I was just setting up github-actions for the first time ever (so exciting) when I noticed that CF doesn't build successfully on clang.
```
build/_deps/cute-src/libraries/cute/cute_tls.h:1248:43: error: arithmetic on a pointer to void
        int w = s2n_send(ctx->connection, data + bytes_written, size - bytes_written, &blocked);
```

The workflow for CF uses gcc, which doesn't complain; or treats that error as a warning instead. It's not a priority for me to fix since I don't actually care about clang, but, it was one of the default compilers that github setup for me.

--- a/libraries/cute/cute_tls.h
+++ b/libraries/cute/cute_tls.h
@@ -1245,7 +1245,7 @@ int tls_send(TLS_Connection connection, const void* data, int size)
 		s2n_blocked_status blocked = S2N_NOT_BLOCKED;
 		int bytes_written = 0;
 		while (bytes_written < size) {
-			int w = s2n_send(ctx->connection, data + bytes_written, size - bytes_written, &blocked);
+			int w = s2n_send(ctx->connection, (const void*)((uintptr_t)data + bytes_written), size - bytes_written, &blocked);
 			if (w >= 0) {
 				bytes_written += w;
 			} else if (s2n_error_get_type(s2n_errno) != S2N_ERR_T_BLOCKED) {

